api/nodecontrol/client.go

```diff
@@ -119,12 +119,38 @@ func (c *ControlAPIClient) FindWorkload(inType, namespace, workloadId string) (*
 	return &envelope.Data, nil
 }
 
+func (c *ControlAPIClient) ListWorkloads(namespace string) ([]nodegen.WorkloadSummary, error) {
+	workloadsInbox := nats.NewInbox()
+
+	var ret []nodegen.WorkloadSummary
+	_, err := c.nc.Subscribe(workloadsInbox, func(m *nats.Msg) {
+		envelope := new(models.Envelope[[]nodegen.WorkloadSummary])
+		err := json.Unmarshal(m.Data, envelope)
+		if err != nil {
+			c.logger.Error("failed to unmarshal workloads response", slog.Any("err", err), slog.String("data", string(m.Data)))
+			return
+		}
+		ret = append(ret, envelope.Data...)
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	err = c.nc.PublishRequest(models.NamespacePingRequestSubject(namespace), workloadsInbox, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	time.Sleep(5 * time.Second)
```

Comment:
>Is this API doing a best effort query without knowledge of how many responses there is suppose to be?

Yes sir

> How long is the average reply expected to take?
I wouldn't think very long, but when Auctioning nodes on the other side of the world, I remember it taking a few seconds....that said, 5s is arbitrary.  I was hoping to find a way to know the number of nodes because that timeout is the single biggest stop to a "snappy" devex right now 😞

Reply:
> I was hoping to find a way to know the number of nodes

Sounds like we may need a heartbeat mechanism onto a shared KV (+TTL in case Nex nodes would die without de-registering). We can think about that one, but will be a follow-up design.